tests/rtl/hardware_test.py

```diff
@@ -340,6 +341,44 @@ def test_hardware_add42_single():
 
     return sdfg
 
+@xilinx_test()
+def test_hardware_axpy_double_pump():
+    # Grab the double pumped AXPY implementation the samples directory
+    spec = importlib.util.spec_from_file_location(
+        "axpy",
+        Path(__file__).parent.parent.parent / "samples" / "rtl" / "axpy_double_pump.py")
+    axpy = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(axpy)
+
+    # Configuration has to use multiple clocks in this sample.
+    old_freq = dace.config.Config.get('compiler', 'xilinx', 'frequency')
+    dace.config.Config.set('compiler', 'xilinx', 'frequency', value='"0:300\\|1:600"')
+
+    # init data structures
+    N = dace.symbol('N')
+    N.set(16)
+    a = np.float32(0)  #np.random.rand(1)[0].astype(np.float32)
```

Comment:
This is a temporary fix, due to the problem of scalars being overwritten by the host as zeros after passing the arguments to the kernel. But, I'll add that as a comment, so that it makes more sense.

Reply:
If it is a free symbol, wont it be passed to the kernel as an argument upon calling the kernel? In regular OpenCL terms. In that case, it would result in the same problem. It could be a constant, so be hardcoded within the generated code, which should work just fine?

Reply:
Yes it is passed as argument. Is the xilinx runtime that is causing this? Plain axpy works fine

Reply:
I've isolated the issue to a vectorization width of 16. The host overwrites the registers with 0's before calling. With a vectorization width of 8, the scaler registers don't get overwritten, so the tests run correctly. I'll investigate some more, but for now the tests run, and no longer pass on 0's